### PR TITLE
Adding sample datasets to be used in the documentation

### LINKED ISSUE #19710: DOC: develop a set of standard example DataFrames for use in docstring examples
Related to https://github.com/pandas-dev/pandas/pull/19704. I didn't find an existing open issue, only a discussion mentioning this in https://github.com/pandas-dev/pandas/pull/16520 (@datapythonista it was actually you then! I didn't realize that :-))

I think it would be good to have a set of standard DataFrames that we reuse throughout our docs (to start with in the docstrings, but we could actually also use a standardized set for the user guide):

- Some small, more "realistic" dataframes would make it is easier to reason about than dummy random data + adds familiarity when reading multiple docstrings
- Makes it easier for contributors to add examples to the docstring as they don't have to invent their own data each time

I don't think there will be "one example dataframe to rule them all", but it would be nice to have a set of them that can cover most of the use cases. 
So we can post some ideas here and discuss them, trying to get to a list.

Side question is whether we want to always define them with code in the docstring, or want to have some example data loading capabilities (eg like seaborn, it examples always start with a `iris = sns.load_dataset("iris")` or other dataset). It can also be a mixture of both of course.

### PR DESCRIPTION
**Pull request by datapythonista**
- [X] closes #19710 
- [ ] tests added / passed
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [X] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
I thought we had an issue about this already, IIRC @mrocklin either made it or commented (was for a slightly different purpose though).

**Comment by jreback** [comment]
We already have lots of data constructors in ``pandas.util.testing``, though these are 'nicer' ones. These would need testing (e.g. do they run), and can be de-privatized (no leading _).

**Comment by jorisvandenbossche** [comment]
> I thought we had an issue about this already, IIRC @mrocklin either made it or commented (was for a slightly different purpose though).

I don't know if there is an older issue, but this is related to what is recently discussed in https://github.com/pandas-dev/pandas/issues/19710 as @datapythonista linked to.

**Comment by jorisvandenbossche** [comment]
> We already have lots of data constructors in pandas.util.testing, though these are 'nicer' ones.

Indeed, I don't think the ones in util.testing are suitable for this, as the exact purpose of those are to be 'nicer' relate-able small dataframes.

**Comment by datapythonista** [comment]
Any more thoughts on this? Knowing which data to use for the examples in the docstrings is the main blocker for the sprint. Any feedback on how you think we can improve this first draft is highly appreciated. Thanks!

**Comment by jorisvandenbossche** [comment]
This needs to be imported in some `__init__` files, as otherwise you cannnot do `pd.io.samples. ...`. Or what would be the intended use in the docs?

File: pandas/io/samples.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
+import numpy
+import pandas
+
+
+def _countries_with_penguins():
```
Comment (jreback) [inline pandas/io/samples.py]: these need to be de-privatized (no leading _)

File: pandas/io/samples.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    """Main dataset to be used in pandas documentation examples."""
+    columns = ('Code', 'Name', 'Capital', 'Continent',
+               'Penguin species', 'Avg. temperature')
+    data = [
+        ('AO', 'Angola', 'Luanda', 'AF', 1, 21.55),
```
Comment (jreback) [inline pandas/io/samples.py]: these need tests

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by datapythonista** [comment]
For what has been discussed in #19710, seems like it probably makes more sense to simply have some ideas on data to be used, but use custom datasets as simple and illustrative as possible depending on each case. So, closing this.

**Pull request closed without merging**